fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1146,6 +1149,9 @@ Future<Version> ReadYourWritesTransaction::getReadVersion() {
 			return resetPromise.getFuture().getError();
 		return RYWImpl::getReadVersion(this);
 	}
+	if( !timeoutActorStarted ) {
```

Comment:
Seeing this now, I realize that we have a bit of a problem with this being implemented in RYW. `getReadVersion` is called implicitly by `get`, `getRange`, etc., but I think only in NativeAPI. For something like this to work, I think we'd have to set the timeout in NativeAPI's getReadVersion, but obviously that's a problem because the timeout lives in RYW.

As it currently stands, I think it's the case that we end up not setting the db timeout at all on a retry that doesn't explicitly call `getReadVersion`.

I guess that means if we want to go with this approach, then we'd need to set the timeout when calling any function that returns a future.

Reply:
Okay, I have something locally that basically adds the logic to ensure the timeout actor is started and adds it to what seems like the right places. I am running into a problem testing it locally, though, which is why I haven’t pushed it yet.

Reply:
I ended up removing that local change and superceded with a different kind of fix in 355f4d5.

Reply:
Wait, wrong commit hash; actually in 22f592c